### ISSUE TITLE
[HOLD PR#35880][$500] IOU - Total amount is only grayed out after revisiting IOU report when creating IOU offline

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#36238]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v.1.4-39-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4295316
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to FAB > Request money > Manual.
3, Create a manual request from a user.
4. Go offline.
5. In 1:1 DM, click on the IOU preview to go to IOU report.
6. Click + > Request money.
7. Create a manual request in local currency.
8. Create another manual request in foreign currency.
9. Go to main chat and return to IOU report.

## Expected Result:
In Step 8, the total amount is grayed out because request with foreign currency is created offline.

## Actual Result:
In Step 8, the total amount is not grayed out.
In Step 9, after revisiting the report, the total amount is grayed out.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/973ddf07-da6a-4650-b8a8-949563a493ca

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01eeb6c185339b648d</li>
        <li>Upwork Job ID: 1755966084970962944</li>
        <li>Last Price Increase: 2024-02-09</li>
    </ul>
</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-split-p2p-chat-groups
CC @gabrielessner

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The total amount of text doesn't grayed out after adding a new request with a different currency while offline.

### What is the root cause of that problem?
The text will be grayed out if `hasUpdatedTotal` returns false.
https://github.com/Expensify/App/blob/04fc8865504eca69e9d03d9023bebd9418baba76/src/components/ReportActionItem/MoneyReportView.tsx#L121-L125

The issue is, `hasUpdatedTotal` depends on transaction data.
https://github.com/Expensify/App/blob/04fc8865504eca69e9d03d9023bebd9418baba76/src/libs/ReportUtils.ts#L4672-L4682

But the component (MoneyReportView) doesn't subscribed to (or have props of) transaction onyx data, so it won't re-render.

### What changes do you think we should make in order to solve the problem?

First of all, we should connect the transaction onyx data.
```
withOnyx({
    transactions: {
        key: ONYXKEYS.COLLECTION.TRANSACTION,
    },
})(MoneyReportView);
```

Then, instead of using `getAllReportTransactions` to get the transaction list, we can pass the `transactions` props to `hasUpdatedTotal`. We can filter before passing it through `selector` or inside `hasUpdatedTotal` itself (I think the selector is better).

_If we concern with the performance, we can create a view wrapper that will calculate the `hasUpdatedTotal` value and pass it down, so MoneyReportView will only re-render if `hasUpdatedTotal` value changes._

**Comment by cubuspl42** [comment]
> This **might** be resolved by #35880 – where we handle the total update when the request with a different currency is made. cc: @cubuspl42 could you please check if this might be related?

I agree that it could have the same root cause. That PR should be merged soon, I gave it a bump!

**Comment by abdulrahuman5196** [comment]
Thanks @cubuspl42. We can have a check back once the PR is merged.

**Comment by sakluger** [comment]
I put the issue on hold for https://github.com/Expensify/App/pull/35880, hopefully that one fixes this!

**Comment by sakluger** [comment]
I also removed Help Wanted temporarily while we wait for that issue.

**Comment by sakluger** [comment]
https://github.com/Expensify/App/pull/35880 has been deployed to prod as of yesterday. I added the 'retest-weekly' label so we can see if this is still reproduceable.

**Comment by sakluger** [comment]
I reached out to #qa on Slack to see if they were able to retest this: https://expensify.slack.com/archives/C9YU7BX5M/p1709589428240689

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by sakluger** [comment]
Looks like the other PR fixed this issue, nice! I'm going to close this issue out, please bump me on Slack if I missed anything.
